### ISSUE TITLE
Screen 12 - Write Changes to Disk

### ISSUE THREAD

**Issue by seb128** [canonical/ubuntu-desktop-installer#37]
The design for screen n°12, Write Changes to Disk

![changes](https://user-images.githubusercontent.com/10974330/110314436-1e37d000-8008-11eb-8031-6a8677fc2cf1.png)

Design details https://people.canonical.com/~platform/desktop/installer/12.%20Write%20Changes%20to%20Disk/

**Comment by fitojb** [comment]
As a translator, the comma splice in the first sentence annoys me.

**Comment by jpnurmi** [comment]
What happens when the user presses back on the next screen? Should this be a popup like in the old installer, because it doesn't seem appropriate to ask writing the same changes again?

**Comment by oSoMoN** [comment]
That's a good question. @long-chun, how does navigation work here?

**Comment by long-chung** [comment]
@oSoMoN At the point of clicking continue; do we do the formatting, or is this at the install page (installation slides)?

**Comment by oSoMoN** [comment]
@long-chung the formatting only happens when we call subiquity's `confirm()` API (@dbungert please correct me if I'm wrong). We're currently calling it when clicking continue, but this is a provisional implementation, the call to `confirm()` will be moved to whenever we really want to start the installation.

**Comment by dbungert** [comment]
This section from the subiquity design is relevant:
https://github.com/canonical/subiquity/blob/main/DESIGN.md#the-server-state-machine

The subiquity controllers can be grouped into two categories:
1) Stuff that Curtin does ("install" models)
2) Stuff that Cloud-init does ("post-install" models)

Filesystem handling is curtin.  
This allows, if desired, the install to get started while we're handling post-install items such as user identity.

That said, please treat a call to /meta/confirm as saying "it is appropriate to format per the storage configuration".

cc @mwhudson

**Comment by mwhudson** [comment]
yeah it's expected that you call /meta/confirm once the install state is NEEDS_CONFIRMATION, basically, which is indicated two ways: via /meta/status and by GET to a unconfigured controller returning x-status: confirm as an http header. It's done this (admittedly confusing) way to support partially interactive installations -- if the keyboard section is the only interactive controller that feeds into the curtin config, then the confirm dialog should be shown immediately after the keyboard screen.

If you wait until the state is NEEDS_CONFIRMATION before calling /meta/confirm then, yes, calling confirm will start the install.
